packages/appium/docs/en/developing/build-doctor-checks.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Building Doctor Checks
+---
+
+The idea of Appium Doctor is to assist users with driver or plugin preconditions setup. Sometimes such
+preconditions might be quite complicated and require non-trivial technical knowledge. Doctor checks,
+which are vanilla Node.js class instances written by extension authors, simplify
+the setup process by automating diagnostics and possible fixes for the found issues. These checks
+might also be interactive to ensure better usage experience.
+
+This tutorial is supposed to be used by plugin or driver authors that would like to help their users
+to deal with complicated setup or configuration steps.
+
+## Adding Doctor Checks
+
+### Typing Requirements
+
+The term `Doctor Check` literally describes a single javascript class instance that implements the
+[IDoctorCheck interface](../../../../types/lib/doctor.ts). The interface defines the following methods:
```

Comment:
This link should be probably changed to https://github.com/appium/appium/blob/master/packages/types/lib/doctor.ts (which doesn't exist yet, but will once this PR is merged). MkDocs won't know how to present a TS file.

Reply:
changed the link